typing/typecore.ml

```diff
@@ -460,6 +460,25 @@ let enter_orpat_variables loc env  p1_vs p2_vs =
   unify_vars p1_vs p2_vs
 
 let rec build_as_type env p =
+  let as_ty = build_as_type_aux env p in
+  match
+    List.find_opt (function
+      | Tpat_constraint _, _, _ -> true
+      | _ -> false
+    ) p.pat_extra
+  with
+  | None -> as_ty
+  | Some (Tpat_constraint cty, _, _) ->
+      begin_def ();
+      let ty = instance cty.ctyp_type in
+      end_def ();
+      generalize_structure ty;
+      (* This call to unify can't fail since the pattern is well typed. *)
+      unify !env (instance as_ty) (instance ty);
+      ty
```

Comment:
I see that you are following the "house style" of not explaining why we do things, but maybe at least a reference to the PR would be helpful for future readers of this code.

Reply:
Just putting a reference to a pull-request doesn't seem that useful. Won't the info show up during `git blame` anyway?

Reply:
I very rarely `git blame` because few tools offer a convenient interface for it (usually I have to slog through whitespace-related changes and file renames before I can find the interesting commit). For me it is definitively more convient to have the information direclty in the source. There are many occurrences of MPR numbers in the compiler source, and I have repeatedly found them very valuable.

Reply:
Meh.
Anyway, as you wish, I pushed a reference.